ait/core/server/handler.py

```diff
@@ -125,26 +131,63 @@ def handle(self, input_data):
         Returns:
             tuple of packet UID and packet data field
         """
-        primary_header_length = 6
+
+        # Convert input_data (packed binary data string) into bytearray
         packet = bytearray(input_data)
+
+        # Check if packet length is at least 7 bytes
+        primary_header_length = 6
         if (len(packet) < primary_header_length + 1):
             ait.core.log.info('CCSDSPacketHandler: Received packet length is less than minimum of 7 bytes.')
             return
-        packet_apid = int(binascii.hexlify(packet[0:2]), 16) & 0x07FF
-        if packet_apid not in self.packet_types:
+
+        # Extract APID from packet
+        packet_apid = str(bin(int(binascii.hexlify(packet[0:2]), 16) & 0x07FF))[2:].zfill(11)
+
+        # Check if packet_apid matches with an APID in the config
+        config_apid = ''
+        for i in self.packet_types:
+            if self.comp_apid(packet_apid, i):
+                config_apid = i
+                break
+        if not config_apid:
```

Comment:
Personally I would move the comparison to each of `self.packet_types` to your `comp_apid` method, and have it return either the `config_apid` it matches or None if it doesn't match any, then replace this chunk with 
```
config_apid = self.comp_apid(packet_apid)
```
But that's just personal preference, this also works.

Reply:
I'll change it to that. I think it looks cleaner too.